src/Microsoft.Security.Utilities.Core/WellKnownRegexPatterns.cs

```diff
@@ -158,7 +158,7 @@ public static string GenerateString(string alphabet, int count)
     private const string SparseAlpha = $"{SparseLowercase}{sparseUppercase}";
     private const string SparseBase62 = $"{SparseAlpha}{SparseDigit}";
     private const string SparseBase64 = $"{SparseBase62}+"; // Forward slash elided.
-    private const string SparseUrlSafeBase64 = $"{SparseBase62}-"; // Underscore elided.
+    private const string SparseUrlSafeBase64 = @$"{SparseBase62}\-"; // Underscore elided.
```

Comment:
>\- [](http://example.com/codeflow?start=64&length=2)

Hyphens in literals that end up in a regex range maybe interpreted in unexpected ways. A hyphen at the end of the sequence will consistent be interpreted as a literal. A hyphen within a character sequence might end up being interpreted as a range operator. In this change we both explicitly escape all hyphens and then also move them to the end of the literal.

This double defensiveness might not be necessary. Previously, the hyphens were placed where they were to indicate their ordering in formats such as base64. In reality, thought, these sequences aren't consistent between base62 and base64 in any case, so there's no substantive value in trying to communicate anything with ordering.